File: 프로그래머스/기초 트레이닝 - Level 0/배열 만들기2-solve.java
import java.util.List;
import java.util.ArrayList; 

class Solution {
    public int[] solution(int l, int r) {
        List<Integer> answer = new ArrayList<>(); 
        
        for (int i = l; i <= r; i++) {
            boolean only5And0 = true;
            for (char num : Integer.toString(i).toCharArray()) {
                if (num != '5' && num != '0') {
                    only5And0 = false;
                    break; 
                }
            }
            
            if (only5And0) answer.add(i);
        }
        
        return !answer.isEmpty() ? answer.stream().mapToInt(i -> i).toArray() : new int[]{-1};
    }
}

File: 프로그래머스/기초 트레이닝 - Level 0/배열 만들기2-solve.py
def solution(l, r):
    answer = [] 
    for i in range(l, r + 1): 
        only_5_and_0 = True
        for num in str(i):
            if num != '5' and num != '0':
                only_5_and_0 = False
                break
        
        if only_5_and_0: 
            answer.append(i) 
            
    return answer if answer else [-1]